public/enh_table_to_ast.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { buildExpression, buildExpressionFunction } from '../../../src/plugins/expressions/public';
+import { getVisSchemas, VisToExpressionAst } from '../../../src/plugins/visualizations/public';
+import { EnhancedTableVisParams } from './components/enhanced_table_vis_options';
+import { EnhancedTableExpressionFunctionDefinition } from './enh_table_fn';
+import { EnhancedTableVisConfig } from './types';
+import { ENH_TABLE_VIS_NAME } from './types'
+
+const buildTableVisConfig = (
+  schemas: ReturnType<typeof getVisSchemas>,
+  visParams: EnhancedTableVisParams
+) => {
+  const metrics = schemas.metric;
+  const buckets = schemas.bucket || [];
+  const visConfig = {
+    dimensions: {
+      metrics,
+      buckets,
+      splitRow: schemas.split_row,
+      splitColumn: schemas.split_column,
+    },
+  };
+
+  if (visParams.showPartialRows && !visParams.showMetricsAtAllLevels) {
+    // Handle case where user wants to see partial rows but not metrics at all levels.
+    // This requires calculating how many metrics will come back in the tabified response,
+    // and removing all metrics from the dimensions except the last set.
+    const metricsPerBucket = metrics.length / buckets.length;
+    visConfig.dimensions.metrics.splice(0, metricsPerBucket * buckets.length - metricsPerBucket);
+  }
+  return visConfig;
+};
```

Comment:
Why do you add this for enhanced_table, whereas you don't define it for "document_table" ?

Reply:
This code was copied from the Data Table visualization `toExpressionAst` from Kibana 7.12. I made this at the beginning of my work and I did not really understood what I was copying. I have removed this code and unified both `enh_table_to_ast.ts`and `doc_table_to_ast.ts` into `to_ast.ts` since now both functions are almost identical